modules/account-lib/src/coin/algo/txnSchema.ts

```diff
@@ -29,3 +31,19 @@ export const BaseTransactionSchema = joi
 
     throw new Error('lastRound cannot be greater than or equal to firstRound');
   });
+
+export const TransferTransactionSchema = joi.object({
+  amount: joi.custom((val) => typeof val === 'number' || typeof val === 'bigint').required(),
+  to: joi
+    .string()
+    .custom((addr) => utils.isValidAddress(addr))
+    .required(),
+  closeRemainderTo: joi.string().optional(),
+});
+export const KeyRegTxnSchema = joi.object({
```

Comment:
we should add some custom validation steps here instead of doing it in the setters.

i.e. check for voteFirst & voteLast constraint and votekey dilution here.

Reply:
I'd imagine theres not much of a difference. I don't see the client building a transaction and not calling the build method immediately.